qualtran/surface_code/azure_cost_model.py

```diff
@@ -64,10 +64,13 @@ def minimum_time_steps(
     """
     c_min = math.ceil(alg.measurements + alg.rotation_gates + alg.t_gates + 3 * alg.toffoli_gates)
     eps_syn = error_budget / 3
-    c_min += math.ceil(
-        alg.rotation_circuit_depth
-        * rotation_model.rotation_cost(eps_syn / alg.rotation_gates).t_gates
-    )
+    if alg.rotation_gates > 0:
+        rotation_cost = rotation_model.prepartion_overhead(eps_syn) + rotation_model.rotation_cost(
+            eps_syn / alg.rotation_gates
+        )
+        c_min += math.ceil(
+            alg.rotation_circuit_depth * (rotation_cost.n_t + 4 * rotation_cost.n_ccz)
+        )
```

Comment:
This would have an effect of multiplying the preparation overhead of rotation with rotation circuit depth, which I don't think we want right? The preparation overhead should be an additive cost to `c_min` instead of a multiplicative cost.

Reply:
fixed